### DISCUSSION TITLE
[Feature]: Add update monitoring (and possibility for auto upgrades) of services and all containers

### DISCUSSION THREAD

**Discussion by peaklabs-dev** [coollabsio/coolify#3162] [Feature Requests]
### Description

Currently, we can manually add a tag to all of our resources and then manually re-deploy all of the resources using the tag. This is not an ideal workflow for keeping containers up to date.

### Minimal Reproduction (if possible, example repository)

Idea of an ideal update workflow (only with coolify), as this workflow currently requires the watchtower docker container:

- [ ] 1. Coolify let us set an interval (via cron) in this set interval coolify will re-pull all images of all containers and if there are updates it will show us an icon in coolify that the container needs to be updated (re-deployed). Do not pull all images at once but 2 at a time for performance reasons --> Update available --> Automatically check for updates for docker images (on the docker hub and services), then if there is an update it will show a badge Update available.
- [ ] It would also be good if we could get a notification via the notification settings we have set in coolify as soon as there are containers that have an update.
- [ ] 2. We can go to a centralized interface and hit update all, which will re-deploy all containers that have a new image ready -> so multiple updates. (better than adding update tag to all containers and going to the tag area)
- [ ] Manual and automated rollback: A rollback feature for normal docker containers and services (other than git) would be amazing, so that when the health check or deployment fails for the new image for the container, the container automatically goes back to the old state it had before (using the old image to deploy the one that was available before, maybe to old image is stored in localhost as long as the container is not healty). 
Example I can depoly the new container for ghost and if the minor PHP update breaks something I can go back to the old image stored in localhost (I hit rollback manully) or if health check fails it rolls back automatically --> Better Image management goes alongside this see: https://github.com/coollabsio/coolify/issues/2499
- [ ] 3. Not as necessary as most likely disabled in production: a switch to automatically update all containers so once a new image is found automatically update
- [ ] Update notifications for coolify panel updates
- [ ] Update notifications for docker image updates

## Examples
- https://github.com/containrrr/watchtower
- https://github.com/getwud/wud

**Comment by texastoland** [comment]
Possible duplicate of #1753 (to consolidate votes) ✅️

**Comment by wottpal** [comment]
Is there already a manual workaround to enable auto-update for containers? Maybe with a weekly cron? 🤔

**Comment by rihards-simanovics** [comment]
I would rather not create a sense of urgency, but I would really love to see some movement on at least the notification side. I had to recently erect a full network separation layer (not to say that this isn't a good practice either way), and lock down the SSH in places on all servers running Coolify management agents. I've seen an uptick in our IDS blocking more and more HTTP-based vulnerability exploit attempts. With the increase in vulnerabilities being detected by LLMs and bad actors exploiting that fact, I grow ever more uncomfortable having no observability on the container services version update side.

As a stop gap measure, I've found that one could use [Diun](https://github.com/crazy-max/diun), but that requires a setup on the individual server level.
